hack/test-integration.sh

```diff
@@ -55,10 +55,14 @@ testexec="${testdir}/${name}.test"
 mkdir -p "${testdir}"
 
 # build the test executable (cgo must be disabled to have the symbol table available)
-pushd "${testdir}" &>/dev/null
-echo "Building test executable..."
-CGO_ENABLED=0 go test -c -tags="${tags}" "${OS_GO_PACKAGE}/${package}"
-popd &>/dev/null
+if [[ -n "${OPENSHIFT_SKIP_BUILD:-}" ]]; then
```

Comment:
Ok, then this is fine.  Are there any other outstanding comments from
anyone?

On Tue, May 31, 2016 at 10:55 PM, Maru Newby <notifications@github.com>
wrote:

> In hack/test-integration.sh
> <https://github.com/openshift/origin/pull/8287#discussion_r65295468>:
>
> > @@ -55,10 +55,14 @@ testexec="${testdir}/${name}.test"
> >  mkdir -p "${testdir}"
> >
> >  # build the test executable (cgo must be disabled to have the symbol table available)
> > -pushd "${testdir}" &>/dev/null
> > -echo "Building test executable..."
> > -CGO_ENABLED=0 go test -c -tags="${tags}" "${OS_GO_PACKAGE}/${package}"
> > -popd &>/dev/null
> > +if [[ -n "${OPENSHIFT_SKIP_BUILD:-}" ]]; then
>
> TEST_ONLY in the case of the core extended tests indicates whether the
> test runner should target an existing cluster or deploy one itself. The
> build behavior isn't controlled via a flag at all. If a given binary is
> missing, hack/build-go.sh is invoked. Otherwise, the binary is not built:
>
> https://github.com/openshift/origin/blob/master/test/extended/setup.sh#L30
>
> This approach requires manual removal of the test binary to ensure a
> binary reflects the latest source. I think it is preferable to rebuild by
> default to minimize the chances of running stale binary, and that the
> extended tests should follow the pattern here rather than the other way
> around.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/8287/files/5eb01fdef0edc21fd5fe921c1e3015ddecd6bbac..e324a228a2ab6c2cc7fdff8881a86dbbdf3b0a78#r65295468>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABG_p3DvSZ4OQFtL0WD70fdIL-SMA86oks5qHPSNgaJpZM4H7j2i>
> .
>